### PR TITLE
dojson: fix lone series number

### PR DESCRIPTION
**Pull request by jacquerie**
Sentry: https://sentry.cern.ch/inspire-sentry/inspire-labs-qa/group/603955/

Closes #1553

### CODE REVIEW

**Comment by jacquerie** [comment]
* First commit does the minimal amount of work to fix an xfailed tests on `acronym`.
* Second commit does the same for `note`.
* Third commit cleans up the style in the DoJSON conferences tests.
* Fourth commit is the meat:

    > Uses the support for DoJSON post-processing functions introduced in
  f4ff61c to remove all series which only have the `number` key.

Ready for review.

File: inspirehep/dojson/conferences/fields/bd1xx.py [hunk 1]
```diff
@@ -29,13 +29,13 @@
...
 from ...utils.geo import parse_conference_address
 
 from inspirehep.utils.helpers import force_force_list
 
 
 @conferences.over('acronym', '^111')
-@utils.for_each_value
 def acronym(self, key, value):
     """Conference acronym."""
     self['date'] = value.get('d')
```
Comment (david-caro) [inline inspirehep/dojson/conferences/fields/bd1xx.py]: shouldn't this iterate over value? (that now is/might be a list of values)

File: inspirehep/dojson/conferences/fields/bd1xx.py [hunk 2]
```diff
@@ -112,42 +115,42 @@ def get_value(value):
...
+    def _get_name(value):
+        return force_single_element(value.get('a'))
+
+    def _get_number(value):
+        n_value = force_single_element(value.get('n'))
+        if n_value and n_value.isdigit():
+            return int(n_value)
+
+    def _last_is_incomplete(series, key):
+        return series and not series[-1].get(key)
```
Comment (david-caro) [inline inspirehep/dojson/conferences/fields/bd1xx.py]: might series be empty at this point? (and cause an IndexError?)

Comment (david-caro) [inline inspirehep/dojson/conferences/fields/bd1xx.py]: No, as you do the series and... xd

File: inspirehep/dojson/conferences/fields/bd1xx.py (hunk 1 above)
Comment (jacquerie) [inline inspirehep/dojson/conferences/fields/bd1xx.py]: Uh, right, I was leaning on the previous tests here but they don't look very reliable. I think I will retract the first two commits.

**Pull request merged**